src/components/Verification/EmailAttributesSelection.tsx

```diff
@@ -0,0 +1,416 @@
+
+		setAttributeData(updatedAttributes);
+
+		await setToLocalStorage(storageKeys.ATTRIBUTE_DATA, JSON.stringify(updatedAttributes));
+	};
+
+
+	const getOrgDetails = async () => {
+		setLoading(true);
+		const orgId = await getFromLocalStorage(storageKeys.ORG_ID);
+		const response = await getOrganizationById(orgId);
+		const { data } = response as AxiosResponse;
+		if (data?.statusCode === apiStatusCodes.API_STATUS_SUCCESS) {
+			const did = data?.data?.org_agents?.[0]?.orgDid;
+
+			if (did.includes(DidMethod.POLYGON) || did.includes(DidMethod.KEY) || did.includes(DidMethod.WEB)) {
+				setW3cSchema(true);
+			}
+			if (did.includes(DidMethod.INDY)) {
+				setW3cSchema(false);
+			}
+		}
+
+		setLoading(false);
+	};
+
+	useEffect(() => {
+		getOrgDetails();
+	}, []);
+
+
+	const handleSubmit = () => {
+		setErrMsg(null);
+	
+		if (w3cSchema) {
+			redirectToAppropriatePage();
+			return;
+		}
+	
+		if (hasInvalidNumberAttributes()) {
+			return;
+		}
+	
+		redirectToAppropriatePage();
+	};
+	
+	const hasInvalidNumberAttributes = (): boolean => {
+		const numberAttributes = attributeData?.filter(
+			(attr) => attr.dataType === 'number' && attr.isChecked
+		);
+	
+		for (const attribute of numberAttributes || []) {
+			if (isInvalidNumberAttribute(attribute)) {
+				return true;
+			}
+		}
+	
+		return false;
+	};
+	
+	const isInvalidNumberAttribute = (attribute: any): boolean => {
+		if (attribute.selectedOption === 'Select' && !attribute.value) {
+			setErrMsg('Condition and value are required');
+			return true;
+		} else if (!attribute.value) {
+			setErrMsg('Value is required');
+			return true;
+		} else if (!attribute.selectedOption) {
+			setErrMsg('Condition is required');
+			return true;
+		}
+	
+		return false;
+	};
+	
+	const redirectToAppropriatePage = () => {
+		window.location.href = w3cSchema
+			? `${pathRoutes.organizations.verification.w3cEmailVerification}`
+			: `${pathRoutes.organizations.verification.emailVerification}`;
+	};
+	
+	const fetchData = async () => {
+	
+		setLoading(true);
+	
+		try {
+			setAttributeData([]);
+	
+			if (w3cSchema) {
+				const getW3CSchemaDetails = await getFromLocalStorage(storageKeys.SELECTED_SCHEMAS);
+				const parsedW3CSchemaDetails = JSON.parse(getW3CSchemaDetails || '[]');
+		
+				if (Array.isArray(parsedW3CSchemaDetails) && parsedW3CSchemaDetails.length > 0) {
+					const allAttributes = parsedW3CSchemaDetails.flatMap(schema => {
+						if (schema.attributes && Array.isArray(schema.attributes)) {
+							return schema.attributes.map(attribute => ({
+								...attribute,
+								schemaName: schema.name,
+								credDefName: '',
+								schemaId: schema.schemaLedgerId,
+								credDefId: ''
+							}));
+						}
+						return [];
+					});
+	
+					const inputArray = allAttributes.map(attribute => ({
+						displayName: attribute.displayName,
+						attributeName: attribute.attributeName,
+						isChecked: false,
+						value: '',
+						condition: '',
+						options: [
```

Comment:
We should create a comman constant to avoid duplicate code

Reply:
Resolved